### PR TITLE
add new recipe for pysbol v0.0.2

### PR DESCRIPTION
**Pull request by kenza12**
**pySBOL** is a SWIG-Python wrapper around [libSBOL](https://github.com/SynBioDex/libSBOL) a module for reading, writing, and constructing genetic designs according to the standardized specifications of the Synthetic Biology Open Language (SBOL).

Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/pysbol/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  sha256: 153432071fdba11ad3612c183f589bee8d9855d088203d5be8832c8026c4e5ec
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install -vv .
+
+requirements:
+  host:
+    - python =3.7
```
Comment (synapticarbors) [inline recipes/pysbol/meta.yaml:19]: Looking at the `setup.py` it looks like this should probably be `python =2.7|>=3.6`

File: recipes/pysbol/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    - pip
+  run:
+    - python =3.7
+    - libxslt
```
Comment (synapticarbors) [inline recipes/pysbol/meta.yaml:23]: I don't see runtime requirements on `requests` or `libxslt` in the `setup.py`.

File: recipes/pysbol/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    - requests
+
+test:
+  imports:
+    - sbol
+
+about:
+  home: https://github.com/SynBioDex/{{ name|lower }}
```
Comment (synapticarbors) [inline recipes/pysbol/meta.yaml:31]: Please don't use jinja templating variables here. Just use the full url.

**Review by synapticarbors — COMMENTED** [review]
I don't fully understand how this gets built so I'm don't know exactly what's going on in the errors. It appears to be looking for the `sbol` package but can't find it; it looks like there is a platform specific directory that isn't getting found. Also please note that there is already a feedstock for the recommended replacement for this package since it's been archived:

https://github.com/conda-forge/pysbol2-feedstock

File: recipes/pysbol/meta.yaml (hunk 1 above)
Comment (kenza12) [reply recipes/pysbol/meta.yaml:34]: Hi @synapticarbors ,

sorry for the long delays, we had to make some changes. This package is working only with python 3.7, the setup has been updated. I’m aware of the availability of pysbol2 but it doesn’t give us the expected results, that’s why we prefer for the moment using pysbol.

Thank you for your feedback and for your work.

File: recipes/pysbol/meta.yaml (hunk 2 above)
Comment (kenza12) [reply recipes/pysbol/meta.yaml]: Hi @synapticarbors ,

we only need **requests** dependency. Thank you.

**Comment by breakthewall** [comment]
Hi, I'm one of the maintainers of this recipe. Thank you!

**Comment by kenza12** [comment]
@conda-forge-admin, please ping conda-forge/<team>

**Comment by kenza12** [comment]
@conda-forge-admin, please ping team

**Comment by kenza12** [comment]
@conda-forge-admin, please ping team

File: recipes/pysbol/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    - 'libstdcxx-ng'
+    - 'libgcc-ng'
+    - lib/libstdc++
+    - lib/libgcc_s.so.1
+    - lib/libstdc++.so.6
+  skip: true  # [win]
+
+requirements:
+  host:
+    - python =3.7
```
Comment (beckermr) [inline recipes/pysbol/meta.yaml:34]: This constraint needs to be in the skip section in build, not here. I think as it stands the recipe will stomp on our global python pinning expressions.

Comment (kenza12) [reply recipes/pysbol/meta.yaml:34]: Hi @beckermr ,

Thank your for your answer.

I'm not sure I understand how to do it. Is that what you mean ?

```
build:
  number: 0
  script: {{ PYTHON }} -m pip install -vv .
  binary_relocation: true
  missing_dso_whitelist:
    - $RPATH/_libsbol.so
    - $RPATH/libxml2-8ed442f7.so.2.6.26
    - $RPATH/libz-a147dcb0.so.1.2.3
    - $RPATH/libxslt-6cbe83bf.so.1.1.17
    - /lib64/libc.so.6
    - /lib64/libm.so.6
    - /lib64/libdl.so.2
    - /lib64/libm.so.6
    - 'libstdcxx-ng'
    - 'libgcc-ng'
    - lib/libstdc++
    - lib/libgcc_s.so.1
    - lib/libstdc++.so.6
  skip: true  # [win] [not py==36]

requirements:
  host:
    - python
    - pip
  run:
    - python
    - requests
```

File: recipes/pysbol/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    - /lib64/libc.so.6
+    - /lib64/libm.so.6
+    - /lib64/libdl.so.2
+    - /lib64/libm.so.6
+    - 'libstdcxx-ng'
+    - 'libgcc-ng'
+    - lib/libstdc++
+    - lib/libgcc_s.so.1
+    - lib/libstdc++.so.6
+  skip: true  # [win]
```
Comment (beckermr) [inline recipes/pysbol/meta.yaml:30]:
```suggestion
  skip: true  # [win or py != 37]
```

File: recipes/pysbol/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    - $RPATH/libxslt-6cbe83bf.so.1.1.17
+    - /lib64/libc.so.6
+    - /lib64/libm.so.6
+    - /lib64/libdl.so.2
+    - /lib64/libm.so.6
+    - 'libstdcxx-ng'
+    - 'libgcc-ng'
+    - lib/libstdc++
+    - lib/libgcc_s.so.1
+    - lib/libstdc++.so.6
```
Comment (beckermr) [inline recipes/pysbol/meta.yaml:29]: you should no need any of this stuff. We cannot repackage binaries. you need to build the code in the recipe

Comment (kenza12) [reply recipes/pysbol/meta.yaml:29]: @beckermr , 
Do you mean build scripts with **build.sh** ? https://docs.conda.io/projects/conda-build/en/latest/resources/build-scripts.html
Do you have an example of a recipe or procedure to handle these binaries please?
Thanks a lot.

**Comment by breakthewall** [comment]
Hi,

It has already been done for sbml2sbol package.

Thank you for your help.

File: recipes/pysbol/meta.yaml (hunk 6 above)
Comment (beckermr) [reply recipes/pysbol/meta.yaml:29]: I don't have an example handy.

**Review by beckermr — CHANGES_REQUESTED** [review]
This recipe needs to build any .so's it has locally. We cannot accept prebuilt binaries.

**Comment by marcelm** [comment]
@kenza12/@breakthewall opened https://github.com/bioconda/bioconda-recipes/pull/31872 in Bioconda without mentioning this PR.

The advice I give over there is to package libSBOL first. It has a CMake option for generating Python wrappers with SWIG, so I assume this is the correct upstream. Note that https://github.com/brsynth/pySBOL/ is maintained by the authors of this PR. Even the repo at https://github.com/SynBioDex/pySBOL, maintained by upstream, apparently contains mostly SWIG-generated stuff, perhaps they only use it for distributing binaries. The correct upstream is very likely libSBOL itself at https://github.com/SynBioDex/libSBOL.

**Comment by carterbox** [comment]
Closing this PR because it is attempting to repackage a pre-compiled binary from a non-official source. The author is free to open a new PR to build libSBOL from source.

**Pull request closed without merging**